pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -959,12 +961,35 @@ public PinotResourceManagerResponse deleteBrokerTenantFor(String tenantName) {
   /**
    * Schema APIs
    */
-  public void addOrUpdateSchema(Schema schema) {
+  public void addOrUpdateSchema(Schema schema, boolean shouldAutoReload, boolean shouldOverride) {
```

Comment:
I would recommend changing this to two method:
`addSchema(Schema schema, boolean override)`
`updateSchema(Schema schema, boolean reload)`

For `updateSchema`, throw exception when schema does not exist, or the change is not backward-compatible

Reply:
What backward-compatible behavior are you referring to? Currently the update schema endpoint (PUT /schemas/{schemaName}) would create the new schema if it did not exist

Reply:
Let me rephrase, when calling `updateSchema()`, if the schema does not exist, or the schema change is not backward-compatible (removing columns, changing data types etc.), throw exception